app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -64,12 +66,12 @@ def feeds
     end
 
     def machine_id(feed)
-      os_resource_for(feed).properties['Machine Id']
+      os_resource_for(hawk_escape_id(feed)).properties['Machine Id']
     end
 
     def os_resource_for(feed)
       with_provider_connection do |connection|
-        os = os_for(feed)
+        os = os_for(hawk_escape_id(feed))
```

Comment:
Isn't this feed being escaped twice? once in os_resource_for and then on [os_for](https://github.com/ManageIQ/manageiq/pull/10048/files#diff-280d41e58c3a9c503d6bd1f110c873b1R85) ?

Wouldn't it be better to escape only prior to a server call or modify the client to escape it?